python/lsst/meas/astrom/astrometry.py

```diff
@@ -255,28 +263,86 @@ def solve(self, exposure, sourceCat):
+            else:
+                refMag = (refCatTemp[loadResult.fluxField]*units.nJy).to_value(units.ABmag)
+                refMagSrcBand = (refCatTemp[refFilterName]*units.nJy).to_value(units.ABmag)
+                refColorDelta = np.nanmedian(refMag - refMagSrcBand)
+                self.log.warning("Adjusting refCat cutoffs for color shift: %s - %s = %.2f}",
+                                 loadResult.fluxField, refFilterName, refColorDelta)
 
         if self.config.doFiducialZeroPointCull:
             nRefCatPreCull = len(refCat)
+            nSelectedSourceCat = len(sourceSelection.sourceCat)
             # Compute rough limiting magnitudes of selected sources
             psfFlux = sourceSelection.sourceCat["base_PsfFlux_instFlux"]
             expTime = exposure.visitInfo.getExposureTime()
-            fiducialZeroPoint = self.config.fiducialZeroPoint[exposure.filter.bandLabel]
+            fiducialZeroPoint = self.config.fiducialZeroPoint[band]
             psfMag = -2.5*np.log10(psfFlux) + fiducialZeroPoint + 2.5*np.log10(expTime)
             sourceMagMin = min(psfMag) - self.config.cullMagBuffer
             sourceMagMax = max(psfMag) + self.config.cullMagBuffer
-            self.log.info("sourceMagMin = %0.3f sourceMagMax = %0.3f", sourceMagMin, sourceMagMax)
-
+            # Account for median ref band - src band color difference...
+            if refColorDelta > self.config.cullMagBuffer:
+                # Start shifting by just half of the median color difference.
+                sourceMagMin += 0.5*refColorDelta
+                sourceMagMax += 0.5*refColorDelta
+                if refColorDelta > 3.0*self.config.cullMagBuffer:
+                    # Shift even furlter if color difference is large compared
+                    # with the cullMagBuffer.
+                    sourceMagMin = np.nanmin((refCat[loadResult.fluxField]*units.nJy).to_value(units.ABmag))
+                    sourceMagMax += 0.5*refColorDelta
+            if refColorDelta < -1.0*self.config.cullMagBuffer:
+                # If the color difference is negative (i.e. sources are fainter
+                # in the observed filter), allow full bright end, and shift to
+                # fainter limit.
+                sourceMagMin = np.nanmin((refCat[loadResult.fluxField]*units.nJy).to_value(units.ABmag))
+                sourceMagMax += refColorDelta
+            self.log.warning("Number of sources = %d;  Number of refs = %d; refs/source = %.2f.",
+                             nSelectedSourceCat, nRefCatPreCull, nRefCatPreCull/nSelectedSourceCat)
+
+            # Include full bright end of reference catalog if there are very
+            # few sources compared to references loaded (this often occurs when
+            # there is a large amount of exctinction and/or scattering from
+            # from bright stars).
+            maxRefToSourceRatio = 20.0  # make config?
```

Comment:
Yes, definitely make this a config.  Probably lots more should be, too, and I think this code is in roughly the right place already, so I'm more more enthusiastic about moving more magic numbers to configs immediately in this package.